test/functional/terminal/tui_spec.lua

```diff
@@ -831,6 +831,22 @@ describe('TUI', function()
       {3:-- TERMINAL --}                                    |
     ]=])
   end)
+
+  it('forwards unknown DCS/OSC sequences to outer terminal', function()
+    local received
+
+    screen._handle_term_unhandled = function(command)
+      received = command
```

Comment:
this won't set a value in the outer scope (except maybe with `debug.setupvalue` ?). Try this instead:

```lua
    local received = {}

    screen._handle_term_unhandled = function(command)
      table.insert(received, command)
    end

```

Reply:
I believe the local handler isn't even called.

```diff
diff --git i/test/functional/terminal/tui_spec.lua w/test/functional/terminal/tui_spec.lua
index db0a7382b..eef43b0c6 100644
--- i/test/functional/terminal/tui_spec.lua
+++ w/test/functional/terminal/tui_spec.lua
@@ -836,6 +836,7 @@ describe('TUI', function()
     local received = {}

     screen._handle_term_unhandled = function(command)
+      print('LOCAL')
       table.insert(received, command)
     end

diff --git i/test/functional/ui/screen.lua w/test/functional/ui/screen.lua
index 25bb2309f..f92cec4ea 100644
--- i/test/functional/ui/screen.lua
+++ w/test/functional/ui/screen.lua
@@ -1538,6 +1538,7 @@ function Screen:_attr_index(attrs, attr)
 end

 function Screen:_handle_term_unhandled(command)
+  print('GLOBAL')
 end

 return Screen
```

When applying this diff, running `TEST_FILTER=outer TEST_FILE=test/functional/terminal/tui_spec.lua make functionaltest` spits out `GLOBAL` 3 times, but no `LOCAL` at all.

So, is this even the correct way to overwrite a function from `screen.lua`?

Reply:
> this won't set a value in the outer scope (except maybe by debug.setupvalue ?). Try this instead:

Upvales are mutable in lua, so this is not a problem.

Rather the problem here is using function instead of method syntax, which is needed even for "local" override. The correct way would be

```
function screen:_handle_term_unhandled(command)
    table.insert(received, command)
end
```

Reply:
Same issue after applying these two fixes: `received` is empty. `screen:_handle_term_unhandled()` still isn't triggered somehow (at least a debug output is not printed).

I pushed the current version in case you want to check it out directly.

Reply:
I've attempted to debug this myself - `term_unhandled_dcs` / `term_unhandled_osc` is called with the right string, but on the lua side we only get an empty string passed into `screen:_handle_term_unhandled`. I suspect there's some marshalling problem.

I suspected that because of the async nature, freeing the string may have been problematic, but I believe we copy the argument before passing it to the lua side, so that's not the problem.

(also - how can we get debug output from tests, without resorting to things like `eq("debug output here", variable_i_want_to_observe)` ?)

Reply:
> (also - how can we get debug output from tests, without resorting to things like `eq("debug output here", variable_i_want_to_observe)` ?)

Currently we don't have a convenient logging function in Lua-land, though it's very much wanted. From C one can use ELOG and that will show up in NVIM_LOG_FILE (which is tail'd by the test runner).